### PR TITLE
HELP-41363: Add HotDesk-Current-ID CCV when device is hotdesked

### PR DESCRIPTION
**Pull request by harenson**
Depends on https://github.com/2600hz/kazoo-configs-kamailio/pull/97 to work.

### CODE REVIEW

File: applications/registrar/src/reg_authn_req.erl [hunk 1]
```diff
@@ -502,3 +503,10 @@ maybe_enforce_security({#auth_user{doc=JObj}=User, Acc}) ->
 -spec maybe_set_encryption_flags({auth_user(), kz_term:proplist()}) -> {auth_user(), kz_term:proplist()}.
 maybe_set_encryption_flags({#auth_user{doc=JObj}=User, Acc}) ->
     {User, encryption_method_map(Acc, JObj)}.
+
+-spec maybe_add_hotdesk_current_id(auth_user()) -> kz_term:proplist().
+maybe_add_hotdesk_current_id(#auth_user{doc=JObj}) ->
```
Comment (jamesaimonetti) [inline applications/registrar/src/reg_authn_req.erl]: Honestly I would just call `hotdesk_ids/1` directly and pattern match:

```erlang
case kzd_devices:hotdesk_ids(JObj, []) of
    [] -> [];
    [Id|_]= -> [{...}]
end.
```

Comment (harenson) [reply applications/registrar/src/reg_authn_req.erl]: Explicit is better than implicit ;)

Comment (jamesaimonetti) [reply applications/registrar/src/reg_authn_req.erl]: Sorry, I meant you don't need the `is_hotdesked/1` call at all. Just the pattern match on `hotdesk_ids/1` will tell you all you need to know.

**Comment by jamesaimonetti** [comment]
@harenson please open a 4.3 PR, cherry-picking the commit to master

**Pull request merged**